### PR TITLE
refactor: remove lifecycle flags

### PR DESCRIPTION
**Pull request by bigopon**
Lifecycle flags have been gradually removed from many places. This is the last step of removing them completely. This involves removing the dispose flag, which is used during deactivation. Previously deactivation can also signal dispose, but now, the dispose call should come separately and manually rather than packed inside deactivation:
Before
```ts
controller.deactivate(..., flags: LifecycleFlags.dispose)
```
After
```ts
controller.deactivate(...)
controller.dispose()
```

### CODE REVIEW

File: packages/router-lite/src/viewport-agent.ts [hunk 1]
```diff
@@ -565,14 +564,17 @@ export class ViewportAgent {
         Batch.start(b1 => {
           tr.run(() => {
             b1.push();
-            return curCA.deactivate(null, controller, LifecycleFlags.dispose);
+            // return curCA.deactivate(null, controller, true);
+            return onResolve(curCA.deactivate(null, controller), () => {
+              curCA.dispose();
+            });
           }, () => {
```
Comment (bigopon) [inline packages/router-lite/src/viewport-agent.ts:571]: @Sayan751 can you help verify this change?

Comment (Sayan751) [reply packages/router-lite/src/viewport-agent.ts:571]: Seems logical

File: packages/router-lite/src/viewport-agent.ts [hunk 2]
```diff
@@ -444,7 +444,7 @@ export class ViewportAgent {
           case 'replace': {
             const controller = this.hostController;
             tr.run(() => {
-              return this.curCA!.deactivate(initiator, controller, LifecycleFlags.dispose);
+              return this.curCA!.deactivate(initiator, controller);
```
Comment (Sayan751) [inline packages/router-lite/src/viewport-agent.ts:447]: We probably also want to dispose here.

File: packages/router-lite/src/viewport-agent.ts [hunk 3]
```diff
@@ -114,7 +114,7 @@ export class ViewportAgent {
         return;
       case State.currIsActive:
         this.logger.trace(`deactivateFromViewport() - deactivating existing componentAgent at %s`, this);
-        return this.curCA!.deactivate(initiator, parent, flags);
+        return this.curCA!.deactivate(initiator, parent);
```
Comment (Sayan751) [inline packages/router-lite/src/viewport-agent.ts:117]: And here as well.

**Pull request merged**